### ISSUE TITLE
Access(ibility) calculations and helper functions

### ISSUE THREAD

**Issue by wklumpen** [r5py/r5py#218]
We need to determine an approach and basic set of access measures to use and what kinds of data pre (or post) processing we want to allow.

We should pull ideas and approaches from:
* This issue on r5r: https://github.com/ipeaGIT/r5r/issues/169
* This `tracc` library from Jeff Allen: https://github.com/jamaps/tracc
* The basic usage examples here: https://ipeagit.github.io/accessibility/

Further work upcoming to compare and implement.

**Comment by wklumpen** [comment]
One more thing to consider from a syntax point of view reading through the r5r issue above: Most 'decay' functions are effectively cumulative probability distributions in some regard (I'm a fan of the 1-normal) which can be drawn directly from `scipy`, and use their language. I did a similar thing over at `mcroute` also to allow for a more generic handling of parameters.

Of course, users can specify their own functions also.

**Comment by wklumpen** [comment]
So, based on my readings of the [Conveyal](https://docs.conveyal.com/learn-more/decay-functions) and a look at the [`accessibility()`](https://ipeagit.github.io/r5r/reference/accessibility.html) function in `r5r` I propose the following, which mimics the structure of r5r:

## Expose the R5 Decay Functions in R5Py
In `r5py` we implement an analog of this accessibility feature which invokes the four decay function options built-in to `r5` (Step Logistic CDF, Exponential Decay, Fixed Exponential Decay, Linear).

I would assume we'd keep the grammar structure from our matrix computation:

```python
access_computer = AccessComputer(
    transport_network,
    origins=origin,
    destinations=points,
    decay_function='linear',
    decay_value=30,
    departure=datetime.datetime(2022,2,22,8,30),
    transport_modes=[TransitMode.TRANSIT, LegMode.WALK]
)

access_computer.compute_access()
```

## Adapt/fork/update `tracc` to generalize computation of transport access measures

Create a separate library used for more generalized or nuanced measures similar to the [accessibility library](https://ipeagit.github.io/accessibility/) created for R

## Side note: Access vs. Accessibility
Personally, I find accessibility to be redundantly long, and more likely confused with physical accessibility in transit conversations. Therefore, I tend to lean towards language of "access" instead of "accessibility". I may be fighting a losing battle but I'll put it out there: Our access objects should not include `ibility` at the end of it :)

**Comment by rafapereirabr** [comment]
I don't have a strong opinion o whether the function in `r5py` should be named `accessibility`or `access` .... but it would be keep the names and functions consistent across `r5r` and `r5py`.

**Comment by HTenkanen** [comment]
Thanks @wklumpen for advancing with this! In general, look good to me! 👍🏻 

I don't either have a strong opinion about the name of the function, but I do think that we should follow similar naming of things between packages which has been one of our design goals. So whether we call it `accessibility` or `access`, I'm fine as long as they would be the same in both packages. 

One other minor comment: This issue now relates to many things which is fine (good for general discussion), but eventually it would be good to make separate issues focusing on the `accessibility()` function and `tracc` related things.

**Comment by wklumpen** [comment]
Okedoke - in that case we should stick with `accessibility` since that matches the r5r lingo. As a devil's advocate (and certainly part of a larger discussion on our copying of r5r) I have another question:

Do we expect many users familiar with r5r to be using r5py? We have been focusing on keeping nomenclature similar because of ease of documentation and because we are familiar with both, but if the goal of r5py is to make it possible for those who don't use R to run R5, perhaps we're thinking about it the wrong way? Food for thought.

**Comment by wklumpen** [comment]
> One other minor comment: This issue now relates to many things which is fine (good for general discussion), but eventually it would be good to make separate issues focusing on the `accessibility()` function and `tracc` related things.

Yes, I'll fire up some separate issues tomorrow for implementing the `accessibility()` function. I'll also reach out to Jeff to see if he would like to keep his library in tact and have us start something new with a bit of a different grammar, or if he's open to substantial changes over there (I'm leaning towards the former).

**Comment by christophfink** [comment]
> Do we expect many users familiar with r5r to be using r5py? We have been focusing on keeping nomenclature similar because of ease of documentation and because we are familiar with both, but if the goal of r5py is to make it possible for those who don't use R to run R5, perhaps we're thinking about it the wrong way? Food for thought.

Absolutely valid points. I guess the most value comes from common branding. OTOH, I do know a few people who use R and Python if not interchangeably, then at least depending on which (scholarly) bubble a project is in.

**Comment by HTenkanen** [comment]
This being said ^, naturally I don't say that there couldn't or shouldn't be any differences between libraries. Naturally there can be and we might end up implementing something different for both libraries depending a bit also on the needs of the users. But I think with these identical functionalities that we have planned to implement, it would be smart to follow same naming. And if there is a good point/rationale in changing a function name, then perhaps it would make sense to change it both in `r5py` and `r5r` at the same time.

**Comment by wklumpen** [comment]
This is a great discussion - and good to hear folks are mixing languages. My general experience is people tend to "pick one" and stick with it (which is why I'm so interested in r5py as I'm not really willing to go through the process of working with R just for one application).

Worth keeping in mind as we discuss - and I think letting the language and community philosophy drive the differences when they are appropriate but otherwise keeping things the same is a good approach.

**Comment by wklumpen** [comment]
Okay so I've made an implementation of this over at [traccess](https://github.com/wklumpen/traccess). I'm not sure what state we want this as for the paper, but I can put together basic documentation and have that ready (it's in a usable spot for basic access but may change rapidly as I develop)

**Comment by christophfink** [comment]
Closing this in favour of #323
